### PR TITLE
Implement  equals and hashcode of the object that is used as the hash key

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
Provide equals and hashCode method of NoRollbackRule,ShutdownHook.DisposablePriorityWrapper

### Ⅱ. Does this pull request fix one issue?

#1431

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java [hunk 1]
```diff
@@ -106,6 +107,23 @@ public int compareTo(DisposablePriorityWrapper disposablePriorityWrapper) {
...
+        @Override
+        public boolean equals(Object other) {
+            if(this == other){
+                return true;
+            }
+            if(!(other instanceof DisposablePriorityWrapper)){
+                return false;
+            }
+            DisposablePriorityWrapper dpw = (DisposablePriorityWrapper)other;
+            return this.priority == dpw.priority;
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java]: disposable?

File: core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java [hunk 2]
```diff
@@ -106,6 +107,23 @@ public int compareTo(DisposablePriorityWrapper disposablePriorityWrapper) {
...
+        @Override
+        public boolean equals(Object other) {
+            if(this == other){
+                return true;
+            }
+            if(!(other instanceof DisposablePriorityWrapper)){
+                return false;
+            }
+            DisposablePriorityWrapper dpw = (DisposablePriorityWrapper)other;
+            return this.priority == dpw.priority && this.disposable == dpw.disposable;
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java]: this.disposable.equals(dpw.disposable)

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/keyword/PostgreSQLKeywordChecker.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import io.seata.rm.datasource.undo.KeywordChecker;
+
+/**
+ * The type PostgreSQL keyword checker.
+ *
+ * @author l81893521
+ * @date 2019/8/8
+ */
+public class PostgreSQLKeywordChecker implements KeywordChecker {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/keyword/PostgreSQLKeywordChecker.java]: revert

File: core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java (hunk 2 above)
Comment (l81893521) [reply core/src/main/java/io/seata/core/rpc/netty/ShutdownHook.java]: already changed

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/keyword/PostgreSQLKeywordChecker.java (hunk 3 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/keyword/PostgreSQLKeywordChecker.java]: already revert

**Pull request merged**